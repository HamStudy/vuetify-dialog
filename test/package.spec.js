import Vue from 'vue'
import Vuetify from 'vuetify'
import VuetifyDialogPlugin from '../src'
import { sleep } from './utils'
Vue.use(Vuetify)

describe('manager', () => {
  let manager

  beforeAll(() => {
    Vue.use(VuetifyDialogPlugin, {
      confirm: {
        actions: ['Foo', 'Bar', 'Baz'],
        icon: false
      }
    })
    manager = Vue.prototype.$dialog
  })

  test('Test confirmation with generic options', async () => {
    const res = await manager.confirm({
      waitForResult: false,
      text: 'Test confirmation with generic options',
      title: 'Title'
    })
    expect(res.vm.$el).toMatchSnapshot()
    await res.close()
    await Vue.nextTick()
    await sleep(1000)
    expect(document.body.innerHTML).toBe('')
  })
})
