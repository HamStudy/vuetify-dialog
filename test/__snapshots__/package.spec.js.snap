// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`manager Test confirmation with generic options 1`] = `
<div
  class="v-dialog__container"
  role="dialog"
>
  <div
    class="v-dialog__content v-dialog__content--active"
    role="document"
    style="z-index: 202; z-index: 202;"
    tabindex="0"
  >
    <div
      class="v-dialog vuedl-layout v-dialog--active dialog-transition-enter dialog-transition-enter-active"
      style="max-width: 450px;"
    >
      <div
        class="vuedl-layout__closeBtn"
      >
        Ã—
      </div>
       
      <div
        class="v-card v-sheet v-sheet--tile theme--light"
      >
        <!---->
         
        <div
          class="v-card__title"
        >
          <h3
            class="headline mb-0"
          >
            Title
          </h3>
        </div>
         
        <div
          class="v-card__text body-1 py-2"
        >
          Test confirmation with generic options
        </div>
         
        <div
          class="v-card__actions"
        >
          <div
            class="spacer"
          />
           
          <div
            class="DialogActions"
          >
            <button
              action-key="Foo"
              class="v-btn v-btn--flat v-btn--text theme--light v-size--default"
              type="button"
            >
              <span
                class="v-btn__content"
              >
                <!---->
                
  Foo
  
                <!---->
              </span>
            </button>
            <button
              action-key="Bar"
              class="v-btn v-btn--flat v-btn--text theme--light v-size--default"
              type="button"
            >
              <span
                class="v-btn__content"
              >
                <!---->
                
  Bar
  
                <!---->
              </span>
            </button>
            <button
              action-key="Baz"
              class="v-btn v-btn--flat v-btn--text theme--light v-size--default"
              type="button"
            >
              <span
                class="v-btn__content"
              >
                <!---->
                
  Baz
  
                <!---->
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
